packages/cli/src/lib/setup/setupUpgrade.ts

```diff
@@ -455,16 +499,16 @@ function Upgrade(options) {
                     }
                     answer = rl.question(
                         `Would you like to ${isUpgrade ? 'upgrade' : 'downgrade'} ${adapter} from @${
-                            ioInstalled.common.version
-                        } to @${version || repoUrl[adapter].version} now? [(y)es, (n)o]: `,
+                            ioInstalled.common!.version
+                        } to @${version || sources[adapter].version} now? [(y)es, (n)o]: `,
```

Comment:
you repeat `sources[adapter]` a lot. I would define a constant and assign `sources[adapter]` to it, then only work with the constant. Makes your life working with TS easier, because it will know that the constant doesn't change - `sources[adapter]` might.